chains/solana/contracts/programs/token-pool/src/lib.rs

```diff
@@ -77,13 +78,30 @@ pub mod token_pool {
 
         let old_authority = ctx.accounts.config.ramp_authority;
         ctx.accounts.config.ramp_authority = new_authority;
-        emit!(RouterUpdated {
+        emit!(RampAuthorityUpdated {
             old_authority,
             new_authority
         });
         Ok(())
     }
 
+    // set_router changes the router program ID. This is used to derive the list
+    // of valid mint/release callers (offramps).
+    pub fn set_router(ctx: Context<SetConfig>, new_router: Pubkey) -> Result<()> {
```

Comment:
could this method be cyclic in some way? Is that a concern?

Reply:
Not sure what you mean exactly about `cyclic` in this case 🤔 
This method may only be invoked by the pool owner, not the router. Then, the router address here is used just to later check if the calling offramp is a valid one or not.

I've checked a bit and we already have another field in the pool though called `ramp_authority` that has the onramp address... And that's always the same as the router address, so we could just reuse it (even though in EVM they are separate concepts, here they are pretty much the same thing....)

Reply:
>And that's always the same as the router address, so we could just reuse it (even though in EVM they are separate concepts, here they are pretty much the same thing....)

I've implemented this change and removed the method (and config field) completely